src/components/panel/panel.js

```diff
@@ -1487,58 +1550,95 @@ MdPanelPosition.prototype._calculatePanelPosition = function(panelEl) {
   var panelWidth = panelBounds.width;
   var panelHeight = panelBounds.height;
 
-  var targetBounds = this._relativeToRect;
+  this._absolute ?
+      this._calculateAbsolutePosition(panelWidth, panelHeight) :
+      this._calculateRelativePosition(panelWidth, panelHeight);
+};
 
-  var targetLeft = targetBounds.left;
-  var targetRight = targetBounds.right;
-  var targetWidth = targetBounds.width;
 
-  switch (this._xPosition) {
-    case MdPanelPosition.xPosition.OFFSET_START:
-      // TODO(ErinCoughlan): Change OFFSET_START for rtl vs ltr.
-      this._right = targetLeft + 'px';
-      break;
-    case MdPanelPosition.xPosition.ALIGN_END:
-      // TODO(ErinCoughlan): Change ALIGN_END for rtl vs ltr.
-      this._right = targetRight + 'px';
-      break;
-    case MdPanelPosition.xPosition.CENTER:
-      var left = targetLeft + (0.5 * targetWidth) - (0.5 * panelWidth)
-      this._left = left + 'px';
-      break;
-    case MdPanelPosition.xPosition.ALIGN_START:
-      // TODO(ErinCoughlan): Change ALIGN_START for rtl vs ltr.
-      this._left = targetLeft + 'px';
-      break;
-    case MdPanelPosition.xPosition.OFFSET_END:
-      // TODO(ErinCoughlan): Change OFFSET_END for rtl vs ltr.
-      this._left = targetRight + 'px';
-      break;
-  }
+/**
+ * Calculates the panel position for absolute based on the created
+ * panel element.
+ * @param {number} panelWidth
+ * @param {number} panelHeight
+ */
+MdPanelPosition.prototype._calculateAbsolutePosition =
+    function(panelWidth, panelHeight) {
+      var windowWidth = window.innerWidth;
```

Comment:
Here's what I currently have for the Angular 2 version (subject to change as we alpha):
https://github.com/angular/material2/blob/master/src/core/overlay/position/global-position-strategy.ts